docs/data/data-grid/row-recipes/DetailPanelExpandOnRowClick.tsx

```diff
@@ -0,0 +1,202 @@
+  randomCommodity,
+} from '@mui/x-data-grid-generator';
+
+function DetailPanelContent({ row: rowProp }: { row: Customer }) {
+  return (
+    <Stack
+      sx={{ py: 2, height: '100%', boxSizing: 'border-box' }}
+      direction="column"
+    >
+      <Paper sx={{ flex: 1, mx: 'auto', width: '90%', p: 1 }}>
+        <Stack direction="column" spacing={1} sx={{ height: 1 }}>
+          <Typography variant="h6">{`Order #${rowProp.id}`}</Typography>
+          <Grid container>
+            <Grid item md={6}>
+              <Typography variant="body2" color="textSecondary">
+                Customer information
+              </Typography>
+              <Typography variant="body1">{rowProp.customer}</Typography>
+              <Typography variant="body1">{rowProp.email}</Typography>
+            </Grid>
+            <Grid item md={6}>
+              <Typography variant="body2" align="right" color="textSecondary">
+                Shipping address
+              </Typography>
+              <Typography variant="body1" align="right">
+                {rowProp.address}
+              </Typography>
+              <Typography
+                variant="body1"
+                align="right"
+              >{`${rowProp.city}, ${rowProp.country.label}`}</Typography>
+            </Grid>
+          </Grid>
+          <DataGridPro
+            density="compact"
+            columns={[
+              { field: 'name', headerName: 'Product', flex: 1 },
+              {
+                field: 'quantity',
+                headerName: 'Quantity',
+                align: 'center',
+                type: 'number',
+              },
+              { field: 'unitPrice', headerName: 'Unit Price', type: 'number' },
+              {
+                field: 'total',
+                headerName: 'Total',
+                type: 'number',
+                valueGetter: (value, row) => row.quantity * row.unitPrice,
+              },
+            ]}
+            rows={rowProp.products}
+            sx={{ flex: 1 }}
+            hideFooter
+          />
+        </Stack>
+      </Paper>
+    </Stack>
+  );
+}
+
+const columns: GridColDef<(typeof rows)[number]>[] = [
+  { field: 'id', headerName: 'Order ID' },
+  { field: 'customer', headerName: 'Customer', width: 200 },
+  { field: 'date', type: 'date', headerName: 'Placed at' },
+  { field: 'currency', headerName: 'Currency' },
+  {
+    field: 'total',
+    type: 'number',
+    headerName: 'Total',
+    valueGetter: (value, row) => {
+      const subtotal = row.products.reduce(
+        (acc: number, product: any) => product.unitPrice * product.quantity,
+        0,
+      );
+      const taxes = subtotal * 0.05;
+      return subtotal + taxes;
+    },
+  },
+];
+
+function generateProducts() {
+  const quantity = randomInt(1, 5);
+  return [...Array(quantity)].map((_, index) => ({
+    id: index,
+    name: randomCommodity(),
+    quantity: randomInt(1, 5),
+    unitPrice: randomPrice(1, 1000),
+  }));
+}
+
+const rows = [
```

Comment:
Any way to use one of the generated datasets? (employees or commodities)

Less lines of code is better.

Reply:
Initially, I tried to refactor all this but was unable to find a more concise way to do it
after seeing that same approach is used in other demos https://mui.com/x/react-data-grid/master-detail/
I ended up doing the same for this demo also.